### PR TITLE
feat: support credentials parameter when generate telemetry client

### PR DESCRIPTION
**Pull request by JiayuW2**
#### Description of change
[//]: # [(What are you trying to fix? What did you change)](feat: support credentials parameter when generate telemetry client)

#### Issue
[//]: # (Having an issue # for the PR is required for tracking purposes. If an existing issue does not exist please create one.)

#### PR reviewer notes
[//]: # (Let us know if there is anything we should focus on.)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/PortingAssistant.Client.Telemetry/TelemetryClient.cs [hunk 1]
```diff
@@ -52,28 +52,31 @@ public Task<AmazonWebServiceResponse> SendAsync(TelemetryRequest request)
...
-            {
-                if (!chain.TryGetAWSCredentials(profile, out awsCredentials))
+                else
                 {
-                    return false;
+                    if (!chain.TryGetAWSCredentials(profile, out awsCredentials))
+                    {
+                        return false;
+                    }
                 }
```
Comment (cslong) [inline src/PortingAssistant.Client.Telemetry/TelemetryClient.cs:77]: I think you need to instantiate the client in this branch of the conditional if we are able to get credentials

**Review by cslong — CHANGES_REQUESTED** [review]

File: src/PortingAssistant.Client.Telemetry/TelemetryClient.cs (hunk 1 above)
Comment (JiayuW2) [reply src/PortingAssistant.Client.Telemetry/TelemetryClient.cs:78]: fix bugs in a new commit and add new tests.

File: tests/PortingAssistant.Client.UnitTests/TelemetryClientFactoryTest.cs [hunk 2]
```diff
@@ -25,5 +26,41 @@ public void EnablingDefaultCredentials_CreatesTelemetryClient()
...
+            Assert.IsNotNull(client);
+        }
+
+        [Test]
+        public void CreatesTelemetryClient_WithoutProfileOrCredentials_Failed()
+        {
+            string profile = null;
+            var telemetryConfig = new TelemetryConfiguration()
+            {
+                InvokeUrl = @"https://8cvsix1u33.execute-api.us-east-1.amazonaws.com/gamma",
```
Comment (cslong) [inline tests/PortingAssistant.Client.UnitTests/TelemetryClientFactoryTest.cs:54]: which endpoint is this, the pasc one?

Comment (JiayuW2) [reply tests/PortingAssistant.Client.UnitTests/TelemetryClientFactoryTest.cs:54]: no this is the old endpoint following other unit tests. Let me try to update it to a dummy.

**Review by cslong — APPROVED** [review]

**Pull request merged**